### PR TITLE
Add support for skipping columns

### PR DESCRIPTION
**Pull request by davidanthoff**
General idea is that if you specify ``nothing`` for a given column in the ``colparsers`` argument, it will just skip that column and not read it in at all. ~~Still needs some work.~~

- [x] Tests
- [x] ``csvread`` for multiple files needs to drop ``nothing`` columns from return value
- [x] UTF8 optimized version
- [x] Docs
- [x] Double check whether the actual parser is correct

EDIT: Alright, this is ready for review. @shashi, would be great if you could take a look, given that this touches a fair number of places in the code base. @joshday I'm not sure how the general multi file stuff interacts with JuliaDB, so it would probably be good if someone from the JuliaDB side could test this branch with important JuliaDB scenarios. Is that something you might be able to do?

Once this is merged and shipped and we have some real world experiences with it, I plan to tackle the real reason I added this in the first place: in the future, when we come across a column that was detected as not-string, and then mid-way through the parsing we realize it actually is a string, we can use this functionality to _just_ reparse that column as a string. I think that should cover the last remaining case where we fail right now if the column type guessing gets something wrong.

### CODE REVIEW

**Comment by davidanthoff** [comment]
Bump. @shashi and @joshday, and chance you might look over this and try whether it works with the JuliaDB use cases?

**Comment by joshday** [comment]
TLDR: Works for single CSVs, but breaks loading multiple CSVs

I tried it out on https://www.kaggle.com/neuromusic/avocado-prices.  It seems to work for one CSV file, although there's a gotcha where `colparsers=Dict(:Large_Bags => nothing)` doesn't work and `colparsers=Dict(Symbol("Large Bags"))` does. I'm not sure if the adding of underscores is on the TextParse or IndexedTables side of things.

For loading multiple CSVs at once, I get an error for any use of `colparsers`, even for things like `colparsers=Dict(1=>Date)`.  Here's what I'm seeing:

```
julia> loadtable(files, colparsers=Dict(1=>nothing))
Error parsing /Users/joshday/datasets/stockdata/Stocks/accp.us.txt
ERROR: BoundsError: attempt to access 0-element Array{Any,1} at index [1]
setindex!(::Array{Any,1}, ::Any, ::Int64) at ./essentials.jl:419
guesscolparsers(::TextParse.VectorBackedUTF8String, ::Array{Any,1}, ::TextParse.LocalOpts{UInt8,UInt8,UInt8}, ::Int64, ::Int64, ::Dict{Int64,Nothing}, ::Type, ::Nothing, ::Array{String,1}, ::Dict{Any,Any}) at /Users/joshday/.julia/dev/TextParse/src/csv.jl:549
#_csvread_internal#52(::Bool, ::Char, ::Char, ::Nothing, ::Type, ::Type, ::Bool, ::Int64, ::Array{String,1}, ::Nothing, ::Int64, ::Array{Any,1}, ::Bool, ::Array{String,1}, ::Array{String,1}, ::OrderedCollections.OrderedDict{Union{Int64, String},Union{Nothing, AbstractArray{T,1} where T}}, ::Int64, ::Dict{Any,Any}, ::Dict{Int64,Nothing}, ::String, ::Int64, ::typeof(TextParse._csvread_internal), ::TextParse.VectorBackedUTF8String, ::Char) at /Users/joshday/.julia/dev/TextParse/src/csv.jl:251
(::getfield(TextParse, Symbol("#kw##_csvread_internal")))(::NamedTuple{(:filename, :rowno, :colspool, :prevheaders, :noresize, :prev_parsers, :samecols, :colparsers),Tuple{String,Int64,OrderedCollections.OrderedDict{Union{Int64, String},Union{Nothing, AbstractArray{T,1} where T}},Array{String,1},Bool,Dict{Any,Any},Array{Any,1},Dict{Int64,Nothing}}}, ::typeof(TextParse._csvread_internal), ::TextParse.VectorBackedUTF8String, ::Char) at ./none:0
(::getfield(TextParse, Symbol("##38#40")){Base.Iterators.Pairs{Symbol,Any,NTuple{7,Symbol},NamedTuple{(:rowno, :colspool, :prevheaders, :noresize, :prev_parsers, :samecols, :colparsers),Tuple{Int64,OrderedCollections.OrderedDict{Union{Int64, String},Union{Nothing, AbstractArray{T,1} where T}},Array{String,1},Bool,Dict{Any,Any},Array{Any,1},Dict{Int64,Nothing}}}},String,Char})(::IOStream) at /Users/joshday/.julia/dev/TextParse/src/csv.jl:117
#open#310(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::getfield(TextParse, Symbol("##38#40")){Base.Iterators.Pairs{Symbol,Any,NTuple{7,Symbol},NamedTuple{(:rowno, :colspool, :prevheaders, :noresize, :prev_parsers, :samecols, :colparsers),Tuple{Int64,OrderedCollections.OrderedDict{Union{Int64, String},Union{Nothing, AbstractArray{T,1} where T}},Array{String,1},Bool,Dict{Any,Any},Array{Any,1},Dict{Int64,Nothing}}}},String,Char}, ::String, ::Vararg{String,N} where N) at ./iostream.jl:369
open at ./iostream.jl:367 [inlined]
#_csvread_f#36 at /Users/joshday/.julia/dev/TextParse/src/csv.jl:114 [inlined]
(::getfield(TextParse, Symbol("#kw##_csvread_f")))(::NamedTuple{(:rowno, :colspool, :prevheaders, :noresize, :prev_parsers, :samecols, :colparsers),Tuple{Int64,OrderedCollections.OrderedDict{Union{Int64, String},Union{Nothing, AbstractArray{T,1} where T}},Array{String,1},Bool,Dict{Any,Any},Array{Any,1},Dict{Int64,Nothing}}}, ::typeof(TextParse._csvread_f), ::String, ::Char) at ./none:0
#csvread#41(::Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:samecols, :colparsers),Tuple{Array{Any,1},Dict{Int64,Nothing}}}}, ::Function, ::Array{String,1}, ::Char) at /Users/joshday/.julia/dev/TextParse/src/csv.jl:149
(::getfield(TextParse, Symbol("#kw##csvread")))(::NamedTuple{(:samecols, :colparsers),Tuple{Array{Any,1},Dict{Int64,Nothing}}}, ::typeof(csvread), ::Array{String,1}, ::Char) at ./none:0
#_loadtable_serial#3(::Char, ::Array{Any,1}, ::Nothing, ::Nothing, ::Nothing, ::Bool, ::Bool, ::typeof(csvread), ::Base.Iterators.Pairs{Symbol,Dict{Int64,Nothing},Tuple{Symbol},NamedTuple{(:colparsers,),Tuple{Dict{Int64,Nothing}}}}, ::typeof(JuliaDB._loadtable_serial), ::UnionAll, ::Array{String,1}) at /Users/joshday/.julia/dev/JuliaDB/src/util.jl:83
(::getfield(JuliaDB, Symbol("##190#193")){Array{Any,1},Base.Iterators.Pairs{Symbol,Dict{Int64,Nothing},Tuple{Symbol},NamedTuple{(:colparsers,),Tuple{Dict{Int64,Nothing}}}},UnionAll})(::Array{String,1}) at ./none:0
do_task(::Dagger.Context, ::Dagger.OSProc, ::Int64, ::Function, ::Tuple{Array{String,1}}, ::Bool, ::Bool, ::Bool) at /Users/joshday/.julia/dev/Dagger/src/scheduler.jl:259
#143 at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/Distributed/src/remotecall.jl:339 [inlined]
run_work_thunk(::getfield(Distributed, Symbol("##143#144")){typeof(Dagger.Sch.do_task),Tuple{Dagger.Context,Dagger.OSProc,Int64,getfield(JuliaDB, Symbol("##190#193")){Array{Any,1},Base.Iterators.Pairs{Symbol,Dict{Int64,Nothing},Tuple{Symbol},NamedTuple{(:colparsers,),Tuple{Dict{Int64,Nothing}}}},UnionAll},Tuple{Array{String,1}},Bool,Bool,Bool},Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}}, ::Bool) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/Distributed/src/process_messages.jl:56
#remotecall_fetch#148(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::Distributed.LocalProcess, ::Dagger.Context, ::Vararg{Any,N} where N) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/Distributed/src/remotecall.jl:364
remotecall_fetch(::Function, ::Distributed.LocalProcess, ::Dagger.Context, ::Vararg{Any,N} where N) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/Distributed/src/remotecall.jl:364
#remotecall_fetch#152(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::Int64, ::Dagger.Context, ::Vararg{Any,N} where N) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/Distributed/src/remotecall.jl:406
remotecall_fetch at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/Distributed/src/remotecall.jl:406 [inlined]
macro expansion at /Users/joshday/.julia/dev/Dagger/src/scheduler.jl:272 [inlined]
(::getfield(Dagger.Sch, Symbol("##13#14")){Dagger.Context,Dagger.OSProc,Int64,getfield(JuliaDB, Symbol("##190#193")){Array{Any,1},Base.Iterators.Pairs{Symbol,Dict{Int64,Nothing},Tuple{Symbol},NamedTuple{(:colparsers,),Tuple{Dict{Int64,Nothing}}}},UnionAll},Tuple{Array{String,1}},Channel{Any},Bool,Bool,Bool})() at ./task.jl:259
Stacktrace:
 [1] compute_dag(::Dagger.Context, ::Dagger.Thunk) at /Users/joshday/.julia/dev/Dagger/src/scheduler.jl:62
 [2] compute(::Dagger.Context, ::Dagger.Thunk) at /Users/joshday/.julia/dev/Dagger/src/compute.jl:25
 [3] #fromchunks#47(::Nothing, ::Int64, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Array{Dagger.Thunk,1}) at /Users/joshday/.julia/dev/JuliaDB/src/table.jl:148
 [4] (::getfield(JuliaDB, Symbol("#kw##fromchunks")))(::NamedTuple{(:output, :fnoffset),Tuple{Nothing,Int64}}, ::typeof(JuliaDB.fromchunks), ::Array{Dagger.Thunk,1}) at ./none:0
 [5] #_loadtable#188(::Nothing, ::Nothing, ::Bool, ::Array{Any,1}, ::Bool, ::Bool, ::Base.Iterators.Pairs{Symbol,Dict{Int64,Nothing},Tuple{Symbol},NamedTuple{(:colparsers,),Tuple{Dict{Int64,Nothing}}}}, ::Function, ::Type, ::Array{String,1}) at /Users/joshday/.julia/dev/JuliaDB/src/io.jl:140
 [6] #_loadtable at ./none:0 [inlined]
 [7] #loadtable#186 at /Users/joshday/.julia/dev/JuliaDB/src/io.jl:63 [inlined]
 [8] (::getfield(JuliaDB, Symbol("#kw##loadtable")))(::NamedTuple{(:colparsers,),Tuple{Dict{Int64,Nothing}}}, ::typeof(loadtable), ::Array{String,1}) at ./none:0
 [9] top-level scope at none:0
```

**Comment by joshday** [comment]
See my edit above.  Looks good on the JuliaDB end!

**Comment by shashi** [comment]
Looks good to me!
